torch/linalg/__init__.py

```diff
@@ -248,42 +268,60 @@
+
+For a square matrix :math:`A`, this is defined as
 
-For a single matrix :attr:`input`, the tensor of eigenvalues `w` and the tensor of eigenvectors
-`V` decompose the :attr:`input` such that `input = V diag(w) V⁻¹`, where `V⁻¹` is the inverse of `V`.
+.. math::
+
+    A = V \operatorname{diag}(L) V^{-1}
+
+This decomposition exists if :math:`A` is diagonalizable (for example, when all its eigenvalues are different).
 
-The eigenvalues are not necessarily sorted.
-For real-valued input the resulting eigenvalues and eigenvectors will always be of complex type.
+The returned decomposition is represented as a namedtuple `(eigenvalues, eigenvectors)`,
+which corresponds to :math:`L`, :math:`V` above. If :attr:`input` is a batch of matrices,
+then `L`, `V` are also batched with the same batch dimensions.
 
-Supports input of float, double, cfloat and cdouble dtypes.
+Supports :attr:`input` of float, double, cfloat and cdouble dtypes.
+The eigenvalues and eigenvectors will always be complex-valued, even if :attr:`input` is real.
 
-.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+.. note:: When :attr:`input` is on a CUDA device, this function synchronizes that device with the CPU.
 
-.. note:: The eigenvalues and eigenvectors are computed using LAPACK's and MAGMA's `geev` routine.
+.. note:: This function is computed using LAPACK's and MAGMA's `geev` routine for
+          CPU and CUDA inputs, respectively.
 
-.. note:: The eigenvectors of matrices are not unique, so any eigenvector multiplied by a constant remains
-          a valid eigenvector. This function may compute different eigenvector representations on
-          different device types. Usually the difference is only in the sign of the eigenvector.
+.. note:: The eigenvectors of a matrix are not unique. Any eigenvector may be multiplied by
+          any non zero number.  As such, the returned eigenvectors are normalized to have norm
+          `1` and largest real component.
+          In particular, the eigenvectors are not continuous functions of the inputs.
 
-.. note:: See :func:`torch.linalg.eigvals` for a related function that computes only eigenvalues.
-          However, that function is not differentiable.
+.. warning:: This function assumes that :attr:`input` is diagonalizable (e.g. when all the
+             eigenvalues are different). If it is not, it will return the correct eigenvalues,
+             but the eigenvectors will be incorrect.
 
 .. warning:: Differentiation support for this function is not implemented yet.
 
-Args:
-    input (Tensor): the `n \times n` matrix or the batch of such matrices of size
-                    `(*, n, n)` where `*` is one or more batch dimensions.
+.. seealso::
 
-Keyword args:
-    out (tuple, optional): tuple of two tensors to write the output to. Default is `None`.
+        :func:`torch.linalg.eigvals` for a related function that computes only eigenvalues.
```

Comment:
"for a related function that" -> ""

would be more consistent with prior templating; also the function is inherently related because it appears in this section

Reply:
"only eigenvalues" -> "only the matrices' eigenvalues"

Reply:
Let's meet in the middle and write "computes only the eigenvalues"?